src/components/Dialogs/CatalogQueryDialog/CatalogOnlineQueryDialogComponent.tsx

```diff
@@ -0,0 +1,305 @@
+import * as React from "react";
+import axios, {CancelTokenSource} from "axios";
+import {action, computed, makeObservable, observable} from "mobx";
+import {observer} from "mobx-react";
+import {AnchorButton, Button, FormGroup, IDialogProps, Intent, InputGroup, MenuItem, NonIdealState, Overlay, PopoverPosition, Spinner} from "@blueprintjs/core";
+import {Tooltip2} from "@blueprintjs/popover2";
+import {IItemRendererProps, Select} from "@blueprintjs/select";
+import {AppStore, CatalogOnlineQueryConfigStore, CatalogDatabase, HelpType, RadiusUnits} from "stores";
+import {DraggableDialogComponent} from "components/Dialogs";
+import {ClearableNumericInputComponent, SafeNumericInput} from "components/Shared";
+import {CatalogSystemType} from "models";
+import "./CatalogOnlineQueryDialogComponent.scss";
+import {ApiService} from "services";
+
+@observer
+export class CatalogQueryDialogComponent extends React.Component {
+    private static readonly DefaultWidth = 550;
+    private static readonly DefaultHeight = 500;
+    private static readonly DBMap = new Map<CatalogDatabase, {type: CatalogSystemType[]; prefix: string}>([
```

Comment:
please move anything interacting with an external API via HTTP to a separate `Service` class